src/openrct2/actions/RideCreateAction.cpp

```diff
@@ -297,7 +297,13 @@ GameActions::Result::Ptr RideCreateAction::Execute() const
     ride->income_per_hour = MONEY64_UNDEFINED;
     ride->profit = MONEY64_UNDEFINED;
     ride->connected_message_throttle = 0;
-    ride->entrance_style = 0;
+
+    ride->entrance_style = OBJECT_ENTRY_INDEX_NULL;
+    if (rtd.HasFlag(RIDE_TYPE_FLAG_HAS_ENTRANCE_EXIT))
+    {
+        ride->entrance_style = gLastEntranceStyle;
```

Comment:
I think it might be better to pass this as a parameter to the game action, this global may be different on other clients, I haven't looked  where this is assigned or when.